lib/sparql.jison

```diff
@@ -703,78 +832,130 @@ SelectClauseVars
     ;
+SelectClauseItem
+    : Var
+    | '(' Expression 'AS' Var ')' -> expression($2, { variable: $4 })
+    ;
+
+// [8]
 SubSelect
     : SelectClauseWildcard WhereClause SolutionModifierNoGroup ValuesClause? -> extend($1, $2, $3, $4, { type: 'query' })
     | SelectClauseVars     WhereClause SolutionModifier        ValuesClause? -> extend($1, $2, $3, $4, { type: 'query' })
     ;
-SelectClauseItem
-    : VAR -> toVar($1)
-    | '(' Expression 'AS' VAR ')' -> expression($2, { variable: toVar($4) })
-    | '(' VarTriple 'AS' VAR ')' -> ensureSparqlStar(expression($2, { variable: toVar($4) }))
-    ;
+
+// [10]
 ConstructQuery
     : 'CONSTRUCT' ConstructTemplate DatasetClause* WhereClause SolutionModifier -> extend({ queryType: 'CONSTRUCT', template: $2 }, groupDatasets($3), $4, $5)
     | 'CONSTRUCT' DatasetClause* 'WHERE' '{' TriplesTemplate? '}' SolutionModifier -> extend({ queryType: 'CONSTRUCT', template: $5 = ($5 ? $5.triples : []) }, groupDatasets($2), { where: [ { type: 'bgp', triples: appendAllTo([], $5) } ] }, $7)
     ;
+
+// [11]
 DescribeQuery
-    : 'DESCRIBE' ( (VAR | iri)+ | '*' ) DatasetClause* WhereClause? SolutionModifier -> extend({ queryType: 'DESCRIBE', variables: $2 === '*' ? [new Wildcard()] : $2.map(toVar) }, groupDatasets($3), $4, $5)
+    : 'DESCRIBE' ( VarOrIri+ | '*' ) DatasetClause* WhereClause? SolutionModifier -> extend({ queryType: 'DESCRIBE', variables: $2 === '*' ? [new Wildcard()] : $2 }, groupDatasets($3), $4, $5)
     ;
+
+// [12]
 AskQuery
     : 'ASK' DatasetClause* WhereClause SolutionModifier -> extend({ queryType: 'ASK' }, groupDatasets($2), $3, $4)
     ;
+
+// [13]
 DatasetClause
-    : 'FROM' 'NAMED'? iri -> { iri: $3, named: !!$2 }
+    : 'FROM' (DefaultGraphClause | NamedGraphClause) -> $2
+    ;
+
+// [14]
+DefaultGraphClause
+    : SourceSelector -> { iri: $1, named: false }
+    ;
+
+// [15]
+NamedGraphClause
+    : 'NAMED' SourceSelector -> { iri: $2, named: true }
+    ;
+
+// [16]
+SourceSelector
+    : iri
     ;
+
+// [17]
 WhereClause
     : 'WHERE'? GroupGraphPattern -> { where: $2.patterns }
     ;
+
+// [18]
 SolutionModifier
     : GroupClause? SolutionModifierNoGroup -> extend($1, $2)
     ;
+
 SolutionModifierNoGroup
     : HavingClause? OrderClause? LimitOffsetClauses? -> extend($1, $2, $3)
     ;
+
+// [19]
 GroupClause
     : 'GROUP' 'BY' GroupCondition+ -> { group: $3 }
     ;
+
+// [20]
 GroupCondition
     : BuiltInCall -> expression($1)
     | FunctionCall -> expression($1)
     | '(' Expression ')' -> expression($2)
-    | '(' Expression 'AS' VAR ')' -> expression($2, { variable: toVar($4) })
-    | VAR -> expression(toVar($1))
+    | '(' Expression 'AS' Var ')' -> expression($2, { variable: $4 })
+    | Var -> expression($1)
     ;
+
+// [21]
 HavingClause
-    : 'HAVING' Constraint+ -> { having: $2 }
+    : 'HAVING' HavingCondition+ -> { having: $2 }
     ;
+
+// [22]
+HavingCondition
+    : Constraint
```

Comment:
There was some deliberate inlining (see long history of this file) because Jison generated file size explodes quite terribly with the number of lines. Can you check what this does to the size?